### PR TITLE
REF: share IntervalIndex.intersection with Index.intersection

### LINKED ISSUE #38299: BUG: find_common_type([interval[f8], interval[i8]]) returns object
I would expect it to return interval[f8], analogous to what we'd get with find_common_type([f8, i8])

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #38299
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This changes the return dtype of IntervalIndex.intersection when that intersection is empty.

### CODE REVIEW

**Comment by jreback** [comment]
nice!

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -1171,3 +1171,15 @@ def __from_arrow__(
             results.append(iarr)
 
         return IntervalArray._concat_same_type(results)
+
+    def _get_common_dtype(self, dtypes: List[DtypeObj]) -> Optional[DtypeObj]:
+        # NB: this doesn't handle checking for closed match
+        if not all(isinstance(x, IntervalDtype) for x in dtypes):
+            return np.dtype(object)
```
Comment (jorisvandenbossche) [inline pandas/core/dtypes/dtypes.py:1178]: This is the wrong return value, see docstring in the base class: https://github.com/pandas-dev/pandas/blob/36c4d5ccf1dbc02b329a55400af55de68415b7e0/pandas/core/dtypes/base.py#L326-L346

**Review by jorisvandenbossche — COMMENTED** [review]
Can you also add tests for concat with interval dtypes? (that will also be affected by this change)

File: pandas/core/dtypes/dtypes.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py:1178]: thanks, will address in follow-up

**Pull request merged**